wgpu-core/src/device/queue.rs

```diff
@@ -1147,7 +1149,9 @@ impl Global {
 
             let device = queue.device.as_ref().unwrap();
 
-            let mut fence = device.fence.write();
+            let snatch_guard = device.snatchable_lock.read();
+
+            let mut fence = device.fence.write(); // snatch(r) -> fence(w)
```

Comment:
I don't understand this comment and the move of the snatch guard. Having worked with wgpu for a while can speculate but it's not clear at all why the ordering is important, please elaborate a bit here.